MegaApiClient/MegaApiClient.cs

```diff
@@ -1311,6 +1322,115 @@ private IEnumerable<int> ComputeChunksSizesToUpload(long[] chunksPositions, long
       }
     }
 
+    private static string GenerateHashcashToken(string challenge)
+    {
+      var parts = challenge.Split(':');
+      if (parts.Length < 4)
+      {
+        throw new ArgumentException("Invalid challenge format");
+      }
+
+      var version = int.Parse(parts[0]);
+      if (version != 1)
+      {
+        throw new ArgumentException("Hashcash challenge is not version 1");
+      }
+
+      var easiness = int.Parse(parts[1]);
+      var tokenStr = parts[3];
+
+      var baseVal = ((easiness & 63) << 1) + 1;
+      var shifts = (easiness >> 6) * 7 + 3;
+      var threshold = baseVal << shifts;
+
+      var token = D64(tokenStr);
+
+      var iterations = 262144;
+      var chunkSize = 48;
+      var prefixSize = 4;
+      var totalSize = prefixSize + iterations * chunkSize;
+
+      var buffer = new byte[totalSize];
+
+      for (var i = 0; i < iterations; i++)
+      {
+        for (var j = 0; j < chunkSize; j++)
+        {
+          buffer[prefixSize + i * chunkSize + j] = token[j];
+        }
+      }
+
+      using (var sha256 = SHA256.Create())
+      {
+        while (true)
+        {
+          var hash = sha256.ComputeHash(buffer);
+          var hashPrefix = BitConverter.ToUInt32(hash, 0);
+
+          if (BitConverter.IsLittleEndian)
+          {
+            hashPrefix = ReverseBytes(hashPrefix);

```

Comment:
```suggestion
            hashPrefix = BinaryPrimitives.ReverseEndianness(hashPrefix);
```
this part was failing for me, so I tried replaced it with the builtin function for reversing the endianess and it seems to work

Reply:
I prefer to not introduce a new dependency here.
@Bl4Cc4t Did you have a chance to investigate why the original code was failing?

@tr4wzified Maybe you can use the following instead
```csharp
if (BitConverter.IsLittleEndian)
{
  Array.Reverse(hash);
}
      
var hashPrefix = BitConverter.ToUInt32(hash, 0);
```

Reply:
@Bl4Cc4t @tr4wzified ? I'm just waiting for your feedback in order to merge this PR and deploy a first version on MyGet.

Reply:
I'll check this out later today

Reply:
Changed it, looks like everything's working here